Hints.py

```diff
@@ -1,29 +1,50 @@
+from __future__ import annotations
+import itertools
+import json
 import logging
 import os
 import random
-from collections import OrderedDict, defaultdict
+import sys
 import urllib.request
-from urllib.error import URLError, HTTPError
-import json
+from collections import OrderedDict, defaultdict
+from collections.abc import Callable, Iterable
 from enum import Enum
-import itertools
+from typing import TYPE_CHECKING, Optional
+from urllib.error import URLError, HTTPError
 
-from HintList import getHint, getMulti, getHintGroup, getUpgradeHintList, hintExclusions, misc_item_hint_table, misc_location_hint_table
-from Item import Item, MakeEventItem
-from Messages import COLOR_MAP, update_message_by_id
+from HintList import Hint, get_hint, get_multi, get_hint_group, get_upgrade_hint_list, hint_exclusions, \
+    misc_item_hint_table, misc_location_hint_table
+from Item import Item, make_event_item
+from Messages import Message, COLOR_MAP, update_message_by_id
 from Region import Region
 from Search import Search
 from TextBox import line_wrap
-from Utils import random_choices, data_path
-
-
-bingoBottlesForHints = (
-    "Bottle", "Bottle with Red Potion","Bottle with Green Potion", "Bottle with Blue Potion",
+from Utils import data_path
+
+if sys.version_info >= (3, 10):
+    from typing import TypeAlias
+else:
+    TypeAlias = str
```

Comment:
Yeah, it's a bit ugly but ¯\_(ツ)_/¯

Reply:
I mean, honestly, we're already bumping the python requirement up, is there any reason to not just go all the way and bump it up to 3.10? (Although that might be more of a question for TreZ)

Reply:
I think when we discussed it in Discord, we eventually settled on not dropping support for 3.9 as it is still a very common version out in the wild. I never went and dropped 3.8 support because I am not sure I saw anything new to 3.9 being particularly relevant.